api_app/serializers.py

```diff
@@ -77,18 +83,183 @@ class _AbstractJobCreateSerializer(rfs.ModelSerializer):
+        tlp = serialized_data.get("tlp", TLP.WHITE).upper()
+
+        # read config
+        analyzer_dataclasses = AnalyzerConfig.all()
+        all_analyzer_names = list(analyzer_dataclasses.keys())
+
+        # run all analyzers ?
+        run_all = len(analyzers_requested) == 0
+        if run_all:
+            # select all
+            selected_analyzers.extend(all_analyzer_names)
+        else:
+            # select the ones requested
+            selected_analyzers.extend(analyzers_requested)
+
+        for a_name in selected_analyzers:
+            try:
+                config = analyzer_dataclasses.get(a_name, None)
+
+                if not config:
+                    if not run_all:
+                        raise NotRunnableAnalyzer(
+                            f"{a_name} won't run: not available in configuration"
+                        )
+                    # don't add warning if run_all
+                    continue
+
+                if not config.is_ready_to_use:
+                    raise NotRunnableAnalyzer(
+                        f"{a_name} won't run: is disabled or unconfigured"
+                    )
+
+                if tlp != TLP.WHITE and config.leaks_info:
+                    raise NotRunnableAnalyzer(
+                        f"{a_name} won't be run because it leaks info externally."
+                    )
+                if tlp == TLP.RED and config.external_service:
+                    raise NotRunnableAnalyzer(
+                        f"{a_name} won't be run because you"
+                        " filtered external analyzers."
+                    )
+                if serialized_data["is_sample"]:
+                    if not config.is_type_file:
+                        raise NotRunnableAnalyzer(
+                            f"{a_name} won't be run because does not support files."
+                        )
+                    if not config.is_filetype_supported(
+                        serialized_data["file_mimetype"]
+                    ):
+                        raise NotRunnableAnalyzer(
+                            f"{a_name} won't be run because mimetype "
+                            f"{serialized_data['file_mimetype']} is not supported."
+                        )
+                else:
+                    if not config.is_type_observable:
+                        raise NotRunnableAnalyzer(
+                            f"{a_name} won't be run because "
+                            "it does not support observable."
+                        )
+
+                    if not config.is_observable_type_supported(
+                        serialized_data["observable_classification"]
+                    ):
+                        raise NotRunnableAnalyzer(
+                            f"{a_name} won't be run because "
+                            "it does not support observable type "
+                            f"{serialized_data['observable_classification']}."
+                        )
```

Comment:
could not we remove this part here considering that this is already handled by the specific versions of `filter_analyzers` of observables/files? so we have not duplicated code